src/infra/core/src/remote_alias_resolver_impl.rs

```diff
@@ -0,0 +1,296 @@
+                    dataset_name,
+                    account_name,
+                ));
+            }
+            let remote_repo_names: Vec<_> = self.remote_repo_reg.get_all_repositories().collect();
+            if remote_repo_names.len() > 1 {
+                return Err(ResolveAliasError::AmbiguousRepository);
+            }
+            if let Some(repository_name) = remote_repo_names.first() {
+                repo_name = repository_name.clone();
+            } else {
+                return Err(ResolveAliasError::EmptyRepositoryList);
+            }
+        }
+        let remote_repo = self.remote_repo_reg.get_repository(&repo_name).int_err()?;
+
+        let access_token_maybe = self
+            .access_token_resolver
+            .resolve_odf_dataset_access_token(&remote_repo.url);
+        if account_name.is_none() {
+            account_name = RemoteAliasResolverApiHelper::resolve_remote_account_name(
+                &remote_repo.url,
+                access_token_maybe.as_ref(),
+            )
+            .await
+            .int_err()?;
+        }
+        let transfer_dataset_name = dataset_name.clone().unwrap_or(
+            self.resolve_remote_dataset_name(
+                local_dataset_handle,
+                &remote_repo.url,
+                access_token_maybe.as_ref(),
+            )
+            .await?,
+        );
+
+        let remote_url = self.combine_remote_url(
+            &remote_repo.url,
+            account_name.as_ref(),
+            &transfer_dataset_name,
+        )?;
+
+        return Ok(RemoteAliasRef::new(
+            remote_url,
+            Some(repo_name),
+            dataset_name,
+            account_name,
+        ));
+    }
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct RemoteAliasResolverApiHelper {}
+
+impl RemoteAliasResolverApiHelper {
+    fn build_headers_map(access_token_maybe: Option<&String>) -> http::HeaderMap {
+        let mut header_map = http::HeaderMap::new();
+        if let Some(access_token) = access_token_maybe {
+            header_map.append(
+                http::header::AUTHORIZATION,
+                http::HeaderValue::from_str(format!("Bearer {access_token}").as_str()).unwrap(),
+            );
+        };
+        header_map
+    }
+
+    // Return account name if remote workspace is in multi tenant mode
+    pub async fn resolve_remote_account_name(
+        server_backend_url: &Url,
+        access_token_maybe: Option<&String>,
+    ) -> Result<Option<odf::AccountName>, GetRemoteAccountError> {
+        let client = reqwest::Client::new();
+        let header_map = Self::build_headers_map(access_token_maybe);
+
+        let workspace_info_response = client
+            .get(server_backend_url.join("info").unwrap())
+            .headers(header_map.clone())
+            .send()
+            .await
+            .int_err()?
+            .error_for_status()
```

Comment:
Have you tested this with plain storage repos like `S3` or local FS? I don't see any logic that would differentiate those from ODF repos.

Even if a repo has HTTP(S) scheme it's not a fact that it's an ODF node. So we likely need some fallback logic for cases when `/info` is not present.